### PR TITLE
Updated agreement IDs and templates, 'minor'

### LINKED ISSUE #87: Update preparation tremplate
Preparation template for datasets might needs some updating for new changes with HUGGO and AIGGO.

As well as some new expectations about missing values, among other things.

### LINKED ISSUE #89: Duplicate manyIDs
Some agreements in manyenviron and manytrade have duplicate manyIDs for different agreements that have different titles.

### LINKED ISSUE #90: Condense and refactor code_states()
Helper functions for `code_states()` have some duplicate functionality that could be removed

### LINKED ISSUE #91: Better standardise treaty titles
Add matches to EC and EFTA in countryregex and improve `standardise_titles()` so that states' names in treaty titles are standardised and matched consistently to reduce errors in manyID generation

### LINKED ISSUE #92: Add update titles and IDs options to export_data()
As datasets in databases are added at different points in time, any changes with ID structures or title standardisation will not automatically happen. This can create issues as unnecessary duplication of IDs when consolidating databases. See issue #89 and #91 for more information.

### PR DESCRIPTION
**Pull request by henriquesposito**
# Description

These minor changes update how agreements are coded and manyID condensed. As well, how treaty titles are standardised and data templates have been updated.

## Package

- Closed #87 by updating data templates for new changes with HUGGO and AIGGO new structure
- Renamed `retrieve_clauses()` to `read_clauses()`

## Correction

- Closed #89 by updating `code_agreements()` and `condense_agreements()`to improve duplicated matching and methods "fuzzy" matching of treatyIDs for the creation of manyIDs
- Closed #90 by updating `code_states()`, previously called `code_parties()`, to work similarly as it did in `{manystates}` to avoid unnecessary overlaps
- Updated country regex data to improve states' ID matching
- Closed #91 by updating `standardising_treaty_titles()` to be faster, more concise, and consistent
- Closed #92 by updating `export_data()` to ask users if treatyIDs, manyIDs, and treaty titles in datasets in a database need to also be updated

# Checklist:

- [x] PR form
  - [x] I have given this pull request an informative title
  - [x] Description above itemizes changes under subtitles, e.g. "## Collection""
  - [x] Any closed, fixed, or related issues are referenced and explained in the description above, e.g. "Fixed #0 by adding A"
  - [x] Package builds on my OS without issues
- [x] PR checks all pass for latest commit
  - [x] Package builds on Mac
  - [x] Package builds on Windows
  - [x] Package builds on Linux
  - [x] CodeCov check: Package improves or maintains good test coverage
  - [x] CodeFactor check: Package improves or maintains good style
- [x] Documentation
  - [x] Any new or modified functions or data have roxygen style documentation in their .R scripts
  - [x] Any longer functions are commented inline so that it is easier to debug in the future
  - [x] PR description above and the NEWS.md file are aligned
  - [x] DESCRIPTION file version is bumped by the appropriate increment (major, minor, patch)

### CODE REVIEW

File: R/code_agreements.R [hunk 1]
```diff
@@ -403,63 +386,51 @@ code_known_agreements <- function(title) {
...
 #' @details Codes acronyms that are 4 to 6 digits long.
-#' For shorter treaty titles, six words or less, acronym
-#' includes first letter of each word.
-#' For longer treaty titles, seven words or more, acronym includes
-#' first letter of first word followed by the number of words in
-#' and title first letter of last word in title.
+#' For shorter treaty titles, six words or less, acronym includes first letter
+#' of each word.
+#' For longer treaty titles, seven words or more, acronym includes first letter
+#' of first word followed by the number of words in and title first letter of
```
Comment (jaeltan) [inline R/code_agreements.R]: wording

File: R/code_lineage.R [hunk 2]
```diff
@@ -23,14 +23,14 @@ code_lineage <- function(title = NULL, database = NULL) {
...
   }
   # code entity and actions for titles
   title <- stringi::stri_trans_general(title, id = "Latin-ASCII")
   entity <- code_entity(title)
   domain <- code_domain(title)
-  parties <- code_parties(title)
+  parties <- code_states(title)
   # Get entity and actions from preamble if missing from title
   if (exists("txt")) {
     entity <- ifelse(is.na(entity), code_entity(txt), entity)
```
Comment (jaeltan) [inline R/code_lineage.R:36]: Not sure where to put this but I think `code_entity()` and the `entity::location_entity()` that it depends on does not seem to be working well

**Review by jaeltan — COMMENTED** [review]
Made some minor comments about spelling and wording, but I think the bigger question I have is how we're going to use functions like code_lineage and code_entity

File: R/code_lineage.R (hunk 2 above)
Comment (henriquesposito) [reply R/code_lineage.R:36]: Thank you Jael! I will look into this

File: tests/testthat/test_export_data.R [hunk 3]
```diff
@@ -1,8 +1,49 @@
-test_that("url is declared", {
-  expect_error(export_data("COW", databse = "states"), "Please use the URL argument to provide a direct webURL to the source of your dataset.")
-})
-
 test_that("url is valid", {
   expect_error(export_data("COW", databse = "states", URL = 5),
```
Comment (jaeltan) [inline tests/testthat/test_export_data.R:2]: spelling error with database

**Review by jaeltan — COMMENTED** [review]
Thanks @henriquesposito for all the work on manypkgs, especially to improve `code_agreements()` and `condense_agreements()`! I will take another look at the fuzzy matching by comparing the consolidated databases using manyIDs and treatyIDs. Other than that, I only have a few minor comments about spelling etc. Please let me know if you need me to work on anything else, including eg. the vignette on treaties, thank you!

**Review by jaeltan — APPROVED** [review]

**Pull request merged**